### PR TITLE
minimal DocC markdown to order Logger.Levels

### LINKED ISSUE #224: Generated documentation doesn't order the Log levels in source order, but alphabetizes them, which causes some confusion while reading it.
### Expected behavior
The generated Documentation (using DocC) describing the log levels would be ordered as they are in the source file

### Actual behavior
The generated Documentation (using DocC) with swift-log presented the levels in alphabetical order

Screenshot attached
Screen Shot 2022-07-19 at 3 02 54 PM

the ordering of which doesn't match the expected [ordering from the source](https://github.com/apple/swift-log/blob/main/Sources/Logging/Logging.swift#L746-L776):

- trace
- debug
- info
- notice
- warning
- error
- critical

I ran into this while working on a Vapor project, and realized the ordering in the generated DocC meant that the descriptions were very "out of whack" for reading with any intention of what the ordering SHOULD be.

swift-log doesn't currently have a DocC catalog - which is where you could put the explicit ordering for this sort of scenario, as least in the case of DocC generated content.

### Steps to reproduce

1. clone out swift-log
2. open Package.swift (w/ Xcode)
3. Product > Build Documentation
4. view Logger > Logger.Level 

### If possible, minimal yet complete reproducer code (or URL to code)

main branch as well as the 1.4.2 branch of swift-log

### SwiftLog version/commit hash

- f2e8667d508016531ba5d044b739eae1ae532a30 (branch `main` HEAD)
- 5d66f7ba25daf4f94100e7022febf3c75e37a6c7 (1.4.2 branch HEAD)

### Swift & OS version (output of `swift --version && uname -a`)

```
swift-driver version: 1.60 Apple Swift version 5.7 (swiftlang-5.7.0.120.1 clang-1400.0.28.1)
Target: arm64-apple-macosx12.0
Darwin Sparrow.local 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:29 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T8101 arm64
```

### PR DESCRIPTION
**Pull request by heckj**
_minimal DocC markdown to order Logger.Levels_

### Motivation:

When viewed in documentation generated by DocC, the enumeration levels
are ordered alphabetically, removintg the implied ordering in the
source. The written overview of the enum relies on that ordering to make
sense.

### Modifications:

Adding a single file that provides structure to Logging/Logger/Levels
within DocC generated documentation.

### Result:

When imported into a project that leverages Swift 5.6 (such as Vapor),
the generated documentation for the levels provided by
Logging/Logger/Levels are displayed in the order intended/expected by
the documentation. Resolves #224.

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-server-bot test this please

**Comment by ktoso** [comment]
Cool, thanks! We worked through this and sadly needed to add a new Package-swift5.6.swift so `docc` can actually find the docc files, but it works well now:

Screenshot 2022-07-20 at 9 06 24

**Comment by ktoso** [comment]
@swift-server-bot test this please

**Review by ktoso — APPROVED** [review]

**Comment by ktoso** [comment]
Heh seems Ci now has issues due to
```
error: emit-module command failed with exit code 1 (use -v to see invocation)[25/28] Emitting module swift_logPackageTests
/code/Tests/LinuxMain.swift:28:1: error: expressions are not allowed at the top level
XCTMain([
^
[26/28] Compiling swift_logPackageTests LinuxMain.swift
/code/Tests/LinuxMain.swift:28:1: error: expressions are not allowed at the top level
XCTMain([
^
error: fatalError
```

on 5.6 builds -- so we may need to #if them out

**Comment by ktoso** [comment]
@swift-server-bot add to allowlist

**Comment by ktoso** [comment]
@swift-server-bot test this please

**Comment by ktoso** [comment]
@swift-server-bot test this please

**Comment by ktoso** [comment]
I'm looking into how to have all swift versions survive this -- will push this over the finish line, thanks for the PR @heckj !

**Comment by ktoso** [comment]
@swift-server-bot test this please

File: Sources/Logging/LogHandler.swift [hunk 1]
```diff
@@ -171,7 +171,8 @@ extension LogHandler {
                     source: String,
                     file: String,
                     function: String,
-                    line: UInt) {
+                    line: UInt)
+    {
```
Comment (tomerd) [inline Sources/Logging/LogHandler.swift:175]: are these formatting changes intentional?

File: scripts/generate_linux_tests.rb [hunk 2]
```diff
@@ -71,6 +71,7 @@ def createExtensionFile(fileName, classes)
     file.write "\n"
 
     for classArray in classes
+      file.write "#if swift(<5.6)\n"
```
Comment (tomerd) [inline scripts/generate_linux_tests.rb:74]: these set of changes seem redundant to me. can you provide more background on what issue these are solving?

**Comment by tomerd** [comment]
thanks for the contribution @heckj 

I am a bit surprised to see the formatting and test generation changes. what issue did you run into that triggered these changes?

File: scripts/generate_linux_tests.rb (hunk 2 above)
Comment (ktoso) [reply scripts/generate_linux_tests.rb:74]: It's a swiftpm regression, I'm talking with Boris about it

**Review by tomerd — CHANGES_REQUESTED** [review]
lets try to find a simpler way to deal with the test generation issues and rollback the formatting changes.

**Comment by ktoso** [comment]
Okey I think I managed to minimize the change and fix the LinuxMain issue with only flags passed to SwiftPM on 5.6+

Have a look folks, I think this'll be good; And we can fix the SwiftPM issue independently as well even on a linux only 5.6 release perhaps.

File: Sources/Logging/LogHandler.swift (hunk 1 above)
Comment (heckj) [reply Sources/Logging/LogHandler.swift]: @tomerd The formatting updates came from the `swiftformat .` command in the `soundness` script. Any future PR updates will need to content with the either a nailed version of swift format for consistency, or update to the current version.

**Comment by ktoso** [comment]
Waiting this out a little bit while we investigate the LinuxMain issue to see if we can solve it more cleanly -- we'll take this over and merge once we have a plan here -- thanks for the PR @heckj !

**Review by tomerd — APPROVED** [review]
this looks much nicer now!

File: Sources/Logging/LogHandler.swift (hunk 1 above)
Comment (ktoso) [reply Sources/Logging/LogHandler.swift]: reverted them -- the issue was mismatch in swift-format versions

**Comment by ktoso** [comment]
Okey we're clear here and we'll look into LinuxMain issues separately -- thank you @heckj !

**Comment by heckj** [comment]
Thanks! Appreciate all the help!

**Pull request merged**